src/rebase.rs

```diff
@@ -0,0 +1,339 @@
+            checkout_builder: None,
+        }
+    }
+
+    /// this will instruct other clients working
+    /// on this `rebase` that you want a quiet rebase experience.
+    /// This is provided for interoperability between Git tools
+    pub fn quiet(&mut self, quiet: bool) -> &mut Self {
+        self.quiet = quiet;
+        self
+    }
+
+    /// Perform an in-memory rebase, will not updated the repository to be in a rebasing-state
+    /// or modify the working directory.
+    pub fn in_memory(&mut self, in_memory: bool) -> &mut Self {
+        self.in_memory = in_memory;
+        self
+    }
+
+    /// The name of the notes reference used to rewrite notes
+    /// for rebased commits when finishing the rebase
+    pub fn rewrite_notes_ref(&mut self, rewrite_notes_ref: Option<CString>) -> &mut Self {
+        self.rewrite_notes_ref = rewrite_notes_ref;
+        self
+    }
+
+    /// Options to control how trees are merged during a `rebase`.
+    pub fn merge_options(&mut self, merge_options: Option<MergeOptions>) -> &mut Self {
+        self.merge_options = merge_options;
+        self
+    }
+
+    /// Options to control how files are written during a `rebase`.
+    pub fn checkout_builder(&mut self, checkout_builder: Option<CheckoutBuilder<'a>>) -> &mut Self {
+        self.checkout_builder = checkout_builder;
+        self
+    }
+
+    /// raw value of options
+    pub unsafe fn raw(&mut self) -> raw::git_rebase_options {
+        let mut checkout_options: raw::git_checkout_options = mem::zeroed();
+        raw::git_checkout_init_options(&mut checkout_options, raw::GIT_CHECKOUT_OPTIONS_VERSION);
+        if let Some(ref mut checkout_builder) = self.checkout_builder {
+            checkout_builder.configure(&mut checkout_options);
+        }
+        let mut merge_options: raw::git_merge_options = mem::zeroed();
+        if let Some(ref opts) = self.merge_options {
+            ptr::copy(opts.raw(), &mut merge_options, 1);
+        }
+
+        raw::git_rebase_options {
+            version: self.version as c_uint,
+            quiet: self.quiet as c_int,
+            inmemory: self.in_memory as c_int,
+            rewrite_notes_ref: ::call::convert(&self.rewrite_notes_ref),
+            merge_options: merge_options,
+            checkout_options: checkout_options,
+        }
+    }
+}
+
+impl<'rebase, 'repo: 'rebase> RebaseOperation<'rebase, 'repo> {
+    /// The type of `rebase` operation
+    pub fn kind(&self) -> RebaseOperationType {
+        unsafe { Binding::from_raw((*self.raw).kind) }
+    }
+
+    /// The commit ID being cherry-picked.  This will be populated for
+    /// all operations except those of type `Exec`
+    pub fn id(&self) -> Oid {
+        unsafe { Binding::from_raw(&(*self.raw).id as *const raw::git_oid) }
+    }
+    /// The executable the user has requested be run.  This will only
+    /// be populated for operations of type `Exec`
+    pub fn exec(&self) -> Option<CString> {
```

Comment:
Typically this crate tries to expose `exec_bytes` and `exec` as non-allocating views into the bytes contained here, could that be done here as well?